### PR TITLE
Deprecate dutch_kp and lovins stemmer as they are removed in Lucene 10

### PR DESCRIPTION
**Pull request by benwtrent**
Lucene 10 has upgraded its Snowball stemming support, as part of those upgrades, two no longer supported stemmers were removed, `KpStemmer` and `LovinsStemmer`. These are `dutch_kp` and `lovins`, respectively.

We will deprecate in 8.16 and will remove support for these in a future version.

### CODE REVIEW

File: docs/changelog/113143.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+pr: 113143
+summary: Deprecate dutch_kp and lovins stemmer as they are removed in Lucene 10
+area: Analysis
+type: deprecation
+issues: []
+deprecation:
+  title: Deprecate dutch_kp and lovins stemmer as they are removed in Lucene 10
+  area: Analysis
+  details: kp, dutch_kp, dutchKp and lovins stemmers are deprecated and will be removed.
+  impact: These stemmers will be removed and will no longer supported.
```
Comment (john-wagster) [inline docs/changelog/113143.yaml:10]: nit: "no longer be supported"

**Review by john-wagster — APPROVED** [review]
LGTM

**Comment by john-wagster** [comment]
@elasticmachine update branch

**Comment by john-wagster** [comment]
@elasticsearchmachine test this please

**Comment by javanna** [comment]
heya the new tests kind of forced me make these stemmers no-op otherwise we'd throw UOE. I made that quick change in https://github.com/elastic/elasticsearch/commit/79d3d6ab60eb7350e767a9d9ff0da754bd6a35cd . I also moved the deprecation warnings to be closer with the creation of the no-op token filter. Not sure whether we want to add tests for this etc.

**Comment by benwtrent** [comment]
@javanna I think that is ok. This PR was focused on getting deprecation warning set up for 8x where behavior is the same just deprecated. I realize this code will have to change for Lucene 10.

**Pull request merged**